portal/src/graphql/portal/ModifyOAuthClientForm.tsx

```diff
@@ -0,0 +1,216 @@
+    onChange: onRedirectUrisChange,
+    onResolveSuggestions: onResolveRedirectUrisSuggestions,
+    defaultSelectedItems: defaultSelectedRedirectUris,
+  } = useTagPickerWithNewTags(clientConfig.redirect_uris);
+
+  const {
+    list: postLogoutRedirectUris,
+    onChange: onPostLogoutRedirectUrisChange,
+    onResolveSuggestions: onResolvePostLogoutUrisSuggestions,
+    defaultSelectedItems: defaultSelectedPostLogoutUris,
+  } = useTagPickerWithNewTags(clientConfig.post_logout_redirect_uris ?? []);
+
+  const errorMessageMap = useMemo(() => {
+    const redirectUrisErrorMessages: string[] = [];
+    const postLogoutRedirectUrisErrorMessages: string[] = [];
+    const violations = parseError(updateAppConfigError);
+    const redirectUrisRegexp = /^\/oauth\/clients\/[^/]*\/redirect_uris(\/[0-9]*)?$/;
+    const postLogoutRedirectUrisRegexp = /^\/oauth\/clients\/[^/]*\/post_logout_redirect_uris(\/[0-9]*)?$/;
+    for (const violation of violations) {
+      switch (violation.kind) {
+        case "minItems":
+          if (redirectUrisRegexp.test(violation.location)) {
+            redirectUrisErrorMessages.push(
+              renderToString(
+                "ModifyOAuthClientForm.redirect-uris.min-items-error",
+                { minItems: violation.minItems }
+              )
+            );
+          }
+          break;
+        case "format":
+          if (redirectUrisRegexp.test(violation.location)) {
+            redirectUrisErrorMessages.push(
+              renderToString(
+                "ModifyOAuthClientForm.redirect-uris.invalid-error"
+              )
+            );
+          }
+          if (postLogoutRedirectUrisRegexp.test(violation.location)) {
+            postLogoutRedirectUrisErrorMessages.push(
+              renderToString(
+                "ModifyOAuthClientForm.post-logout-redirect-uris.invalid-error"
+              )
+            );
+          }
+          break;
+        default:
+          break;
+      }
+    }
+
+    return {
+      redirectUris: defaultFormatErrorMessageList(redirectUrisErrorMessages),
+      postLogoutRedirectUris: defaultFormatErrorMessageList(
+        postLogoutRedirectUrisErrorMessages
+      ),
+    };
+  }, [updateAppConfigError, renderToString]);
+
+  useEffect(() => {
+    onClientConfigChange(
+      constructClientConfigState(
+        clientConfig.client_id,
+        accessTokenLifetime,
+        refreshTokenLifetime,
+        redirectUris,
+        postLogoutRedirectUris
+      )
+    );
+  }, [
+    clientConfig.client_id,
+    onClientConfigChange,
+
+    accessTokenLifetime,
+    refreshTokenLifetime,
+    redirectUris,
+    postLogoutRedirectUris,
+  ]);
+
+  return (
+    <section className={cn(styles.root, className)}>
+      <TextField
+        className={styles.inputField}
+        label={renderToString("ModifyOAuthClientForm.name-label")}
+        value={clientName}
+        onChange={onClientNameChange}
+        required={true}
+      />
+      <TextField
```

Comment:
Can we use `type="number"` input field for lifetime inputs?

Reply:
I considered using `type="number"` but there is a several problems.

As `type="number"` allows symbol like `.`, `-`, so I think intercepting the onChange function is simpler, avoiding the need to do extra validation to ensure number is positive whole number. 

Also, if I add `type="number"`, then onChange is fired only if `Number()` of value is not `NaN`, so user will be able to input something like `------` and the state will not change.

Reply:
Okay I see. I suppose we can keep using text type input for now.